tests/node-selector_test.go

```diff
@@ -0,0 +1,327 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/litmuschaos/chaos-operator/pkg/apis/litmuschaos/v1alpha1"
+	"github.com/litmuschaos/litmus-e2e/pkg"
+	"github.com/litmuschaos/litmus-e2e/pkg/environment"
+	"github.com/litmuschaos/litmus-e2e/pkg/types"
+	"github.com/litmuschaos/litmus-e2e/utils"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/klog"
+)
+
+func TestGoNodeSelector(t *testing.T) {
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "BDD test")
+}
+
+var _ = Describe("BDD of Node selector", func() {
+	Context("Test for target node selection by node name", func() {
+		It("Execute the node-cpu-hog experiment using a valid TARGET_NODES value", func() {
+			testsDetails := types.TestDetails{}
+			clients := environment.ClientSets{}
+			chaosEngine := v1alpha1.ChaosEngine{}
+			chaosExperiment := v1alpha1.ChaosExperiment{}
+
+			//Getting kubeConfig and Generate ClientSets
+			By("[PreChaos]: Getting kubeconfig and generate clientset")
+			err := clients.GenerateClientSetFromKubeConfig()
+			Expect(err).To(BeNil(), "Unable to Get the kubeconfig, due to {%v}", err)
+
+			testsDetails.ExperimentName = "node-cpu-hog"
```

Comment:
why do we need to setup this here the GetENV function is already doing this in line 43.

Reply:
This is because the logs would otherwise show an empty experiment name for the following operation after the 37th line. Shall I remove it?

Reply:
We can swap line 43 with 42 then it will first set the variable then print the logs and you will get the value. We don't need to initialise the struct variable here

Reply:
Fixed